### ISSUE TITLE
Publicly shareable developer profile URLs

### ISSUE THREAD

**Issue by joemasilotti** [joemasilotti/railsdevs.com#646]
The goal of this issue is to give developers a way to privately share their profile with others with all hidden information revealed. For example, when using their RailsDevs profile to share with a hiring manager.

A PR addressing this issue should:

- [ ] Add a new column to the database, `developers.public_profile_key`
- [ ] Unhide hidden information on a developers profile page if this key matches from the `key` param (e.g. `/developers/abc123?key=xyz987`)
- [ ] Add a new button next to “Edit” when viewing **your own** developer profile titled “Share”. When clicked:
  - [ ] Generate the public profile key via Secure Random (if not already generated)
  - [ ] Copy the full URL (with `key` param) to the clipboard

~This is blocked on #645.~ This is no longer blocked.

**Comment by sarvaiyanidhi** [comment]
Hi @joemasilotti I would like to work on this feature... Can you please assign it to me

**Comment by joemasilotti** [comment]
Sure thing - all yours! Please also hide the button unless the new feature flags is enabled.

**Comment by sarvaiyanidhi** [comment]
Sure Joe.. I will hide button by checking feature flags

Implementation is still in progress but I have one query.

So in order to implement clipboard copy feature I was checking existing implementation to use same code as done in Admin for Copy button but when I click on it it is throwing me below error when I am checking locally in chrome browser

`clipboard.js:11 DOMException: Support for multiple ClipboardItems is not implemented.`

Should I look for alternate approach to get clipboard functionality?

**Comment by joemasilotti** [comment]
Oh no! It looks like that is actually a bug in the existing code and not related to your changes. I just double checked in Chrome and saw the same issue.

If you can, please test in Safari. I'll create a new issue to address what you found.

**Comment by joemasilotti** [comment]
Darn, sorry about that! Feel free to comment out the JavaScript code for your testing. I can pull and run on my machine with Safari to make sure it all works.

**Comment by sarvaiyanidhi** [comment]
Thanks.. Will as well try to see if clipboard issue can be fixed.

**Comment by joemasilotti** [comment]
Closed via #665.
